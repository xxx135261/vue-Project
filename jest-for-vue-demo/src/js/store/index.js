import Vuex from 'vuex'
import Vue from 'vue'
import axios from 'axios'

Vue.use(Vuex);

export const state = {
    msgList: [],
    maxListSize: 1000,
    echartConfig: {
        tooltip: {
            trigger: 'item',
            backgroundColor:'#528DFF',
            color:'#fff',
            formatter: "{a} <br/>{b} : {c} ({d}%)"
        },
        toolbox: {
            show: true,
            feature: {
                dataView: {
                    readOnly: true,
                    iconStyle: {
                       	normal: {
                            borderColor: '#528DFF'
                       	}
                    },
                },
                restore: {
                	iconStyle: {
                       	normal: {
                            borderColor: '#528DFF'
                       	}
                    },
                },
                saveAsImage: {
                    name: '行业配置',
                    iconStyle: {
                       	normal: {
                            borderColor: '#528DFF'
                       	}
                    },
                }
            },
            right: "16px"
        },
        legend: {
            orient: 'vertical',
            left: 'left',
            data: ['农业', '计算机', '其他']
        },
        series: [{
            type: 'pie',
            radius: '55%',
            center: ['50%', '68%'],
            data: [
                {value: 35, name: '农业'},
                {value: 10, name: '计算机'},
                {value: 55, name: '其他'}
            ]
        }],
        grid: {
            left: 'center'
        }
    }
}


export const getters = {
    msgAmount: state => state.msgList.length
}

export const actions = {
    getMsgFromRemote: ({commit}) => {
        axios
            .get('https://www.easy-mock.com/mock/5b765aef4d2b8f332fda9656/msgpool/mock-msg')
            .then(resp => {
                commit('pushMsg', {
                    msg: resp.data
                })
            })
    }
}

export const mutations = {
    pushMsg: (state, payload) => {
        let msgList = state.msgList, currLen = msgList.length;
        if (currLen === state.maxListSize) {
            msgList.splice(0, 1);
        }
        msgList.push(payload.msg);
    }
}

export default new Vuex.Store({
    state,
    getters,
    mutations,
    actions
});