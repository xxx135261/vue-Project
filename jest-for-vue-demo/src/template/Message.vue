<template>
    <li @click="handleClick">
      <div class="block msg-font round-border border-shadow">
        <span>{{msg.name}}：</span>
        <span>{{msg.message}}</span>
      </div>
    </li>
</template>
<script>
export default {
  name: "Message",
  props: {
    msg: {
      type: Object,
      required: true
    }
  },
  methods: {
    handleClick() {
      this.$emit('itemClicked');
    }
  }
};
</script>
<style scoped>
li {
  cursor: pointer;
}
.block {
  width: 100%;
  min-height: 40px;
  background-color: #f4f8fb;
  margin: 8px 0px;
  padding: 3px;
}
.round-border {
  border: 1px solid #f4f8fb;
  border-radius: 6px;
}
.border-shadow {
  box-shadow:2px 2px 4px #1d262c
}
.msg-font {
  color: #45799d;
  font-family: Arial, Helvetica, sans-serif, "Microsoft Yahei", "微软雅黑"
}
</style>


