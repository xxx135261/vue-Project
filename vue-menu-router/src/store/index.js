import Vue from 'vue'
import Vuex from 'vuex'
Vue.use(Vuex)
const state = {
  menulist:[]
}
const	getters = {
	menulist: state => {
		return state.menulist;
	}
}
const	mutations = {
	setMenulist:(state,payLoad) => {
		state.menulist = payLoad
	}
}
export default new Vuex.Store({
	state,
	getters,
	mutations
})
