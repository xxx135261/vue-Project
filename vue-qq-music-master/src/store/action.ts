import {
    SET_HISTORY_KEY,
    SAVE_LOVE,
    GET_SONG,
    SHOW_PLAYER,
    CANCEL_PLAYER,
    SET_PLAYING_STATE
} from './mutation-types';
import { ActionTree, Action } from 'vuex';

// 存储历史记录
const setLocalStorageData: Action<any, any> = ({ commit }, payload) => {
    commit(SET_HISTORY_KEY, payload);
};

const getSong: Action<any, any> = ({ commit }, payload) => {
    commit(GET_SONG, payload);
};

const showPlayer: Action<any, any> = ({ commit }) => {
    commit(SHOW_PLAYER);
};
// 存储收藏的歌曲（触发mutation中的SAVE_LOVE事件
const setLocalStorageData2: Action<any, any> = ({ commit }, payload) => {
    commit(SAVE_LOVE, payload);
};

const cancelPlayer: Action<any, any> = ({ commit }) => {
    commit(CANCEL_PLAYER);
};

const setPlayState: Action<any, any> = ({ commit }, payload) => {
    commit(SET_PLAYING_STATE, payload);
};

const actions: ActionTree<any, any> = {
    setLocalStorageData,
    setLocalStorageData2,
    getSong,
    showPlayer,
    cancelPlayer,
    setPlayState
};
export default actions;
