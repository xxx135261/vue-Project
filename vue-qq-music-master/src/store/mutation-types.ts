// 缓存搜索历史
export const SET_HISTORY_KEY = 'SET_HISTORY_KEY';
// 缓存收藏的歌曲
export const SAVE_LOVE = 'SAVE_LOVE';
// 获取歌曲
export const GET_SONG = 'GET_SONG';
// 显示 player
export const SHOW_PLAYER = 'SHOW_PLAYER';
// 不显示 player
export const CANCEL_PLAYER = 'CANCEL_PLAYER';
// 播放状态（true/false)
export const SET_PLAYING_STATE = 'SET_PLAYING_STATE';
