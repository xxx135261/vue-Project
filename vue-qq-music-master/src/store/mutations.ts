import {
    SET_HISTORY_KEY,
    SAVE_LOVE,
    GET_SONG,
    SHOW_PLAYER,
    CANCEL_PLAYER,
    SET_PLAYING_STATE
} from './mutation-types';
import { Mutation, MutationTree } from 'vuex';

const mutations: MutationTree<any> = {
    // 缓存搜索历史
    [SET_HISTORY_KEY](state, keyword) {
        localStorage.setItem('SET_HISTORY_KEY', keyword);
    },
    // 缓存搜索历史
    [SAVE_LOVE](state, loves) {
        localStorage.setItem('SAVE_LOVE', loves);
    },
    // 获取歌曲
    [GET_SONG](state, payload) {
        state.song = payload;
        state.song.singer.forEach(singer => {
            state.artist = singer.name;
        });
    },
    // 显示 player
    [SHOW_PLAYER](state) {
        state.isShowPlayer = true;
    },
    // 不显示 player
    [CANCEL_PLAYER](state) {
        state.isShowPlayer = false;
    },
    // 播放状态（true/false)
    [SET_PLAYING_STATE](state, flag) {
        state.isPlay = flag;
    }
};

export default mutations;
