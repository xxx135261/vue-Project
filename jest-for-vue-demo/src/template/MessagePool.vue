<template>
    <div class="pool">
      <ul>
          <Message @itemClicked="handleClick" v-for="(msg, index) in msgList" :key="index" :msg="msg"></Message>
      </ul>
    </div>
</template>
<script>
import Message from "./Message";

export default {
  name: "MessagePool",
  components: { Message },
  data() {
    return {
      timer: null,
      maxVisibleLength: 10,
      data: null
    };
  },
  computed: {
    msgList() {
      let msgList = this.$store.state.msgList,
          len = msgList.length;
      let start = len - this.maxVisibleLength;
      return msgList.slice(start >= 0 ? start : 0, msgList.length);
    }
  },
  mounted() {
    this.start();
  },
  destroyed() {
    this.stop();
  },
  methods: {
    genMsg() {
      this.$store.dispatch('getMsgFromRemote');
    },
    stop() {
      if (this.timer) {
        console.log("Timer is stopped.");
        clearInterval(this.timer);
        this.timer = null;
      }
    },
    resume() {
      if (!this.timer) {
        this.start();
      }
    },
    start() {
      if (this.timer) {
        return;
      }
      this.timer = setInterval(() => {
        this.genMsg();
      }, 2000);
      console.log("Timer is ticking.");
    },
    handleClick() {
      this.timer != null ? this.stop() : this.resume();
    }
  }
};
</script>
<style>
ul {
  list-style: none;
  width: calc(100% - 24px);
  padding: 0;
  margin: 0 8px;
}
.pool {
  width: 800px;
  min-height: 580px;
  border: 1px solid #0c6188;
  border-radius: 5px;
  background: #0c6188;
}
</style>


