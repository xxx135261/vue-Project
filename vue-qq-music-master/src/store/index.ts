import Vue from 'vue';
import Vuex from 'vuex';
import mutations from './mutations';
import actions from './action';
Vue.use(Vuex);

export const state = {
    isShowPlayer: false, // 是否显示 player
    song: {}, // 歌的信息
    artist: '', // 歌的作者
    isPlay: false, // 是否播放
    msg: 'hello world!'
};
export default new Vuex.Store({
    state,
    actions,
    mutations
});
