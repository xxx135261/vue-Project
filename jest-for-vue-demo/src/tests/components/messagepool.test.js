import MessagePool from './../../template/MessagePool'
import Message from './../../template/Message'
import {mount, createLocalVue} from '@vue/test-utils'
import Vuex from 'vuex'
import {state, getters} from './../../js/store'

const localVue = createLocalVue();

localVue.use(Vuex);

describe('MessagePool', () => {
    let wrapper;
    let store;
    let GET_MSG_FROM_REMOTE, PUSH_MSG;

    beforeEach(() => {
        GET_MSG_FROM_REMOTE = jest.fn();
        PUSH_MSG = jest.fn();
        
        store = new Vuex.Store({
          state,
          getters,
          mutations: {
            pushMsg: PUSH_MSG
          },
          actions: {
              getMsgFromRemote: GET_MSG_FROM_REMOTE
          }        
        });
        wrapper = mount(MessagePool, {
            mocks: {
                $store: store
            }
        });
    })

    /* structure */
    test('has expected html', () => {
        expect(wrapper.vm.$el).toMatchSnapshot();
    })

    /* structure */
    test('has two message', () => {
        GET_MSG_FROM_REMOTE = jest.fn();
        PUSH_MSG = jest.fn();
        
        store = new Vuex.Store({
          state: {
            msgList: [{
                "name": "mock-name1",
                "message": "mock-msg1"
            }, {
                "name": "mock-name2",
                "message": "mock-msg2"
            }],
            maxListSize: 1000
          },  
          getters,
          mutations: {
            pushMsg: PUSH_MSG
          },
         actions: {
              getMsgFromRemote: GET_MSG_FROM_REMOTE
          }        
        });
        wrapper = mount(MessagePool, {
            mocks: {
                $store: store
            }
        });
        expect(wrapper.findAll('li').length).toBe(2);
    })

    test('is timer started', () => {
        expect(wrapper.vm.timer).not.toBeNull();
    })

    describe('test stop and resume', () => {
        test('is timer stopped', () => {
            wrapper.vm.stop();
            expect(wrapper.vm.timer).toBeNull();
        })

        test('is timer resume', () => {
            wrapper.vm.stop();
            wrapper.vm.resume();
            expect(wrapper.vm.timer != null).toBe(true);
        })
    })

    /* custom event */
    describe('test event', () => {
        beforeEach(() => {
            store = new Vuex.Store({
                state: {
                    msgList: [{
                        "name": "mock-name1",
                        "message": "mock-msg1"
                    }]
                },
                getters,
                mutations: {
                  pushMsg: PUSH_MSG
                },
                actions: {
                    getMsgFromRemote: GET_MSG_FROM_REMOTE
                }        
              });
              wrapper = mount(MessagePool, {
                  mocks: {
                      $store: store
                  }
              });
        })

        test('test the custom event itemClicked is fired', () => {
            let stub = jest.fn();
            wrapper.setMethods({
                handleClick: stub
            });
            wrapper.vm.$on('itemClicked');
            wrapper.find('li').trigger('click');
            expect(stub).toBeCalled();
        })

        test('test the custom event itemClicked can be emitted', () => {
            let stub = jest.fn();
            wrapper.setMethods({
                handleClick: stub
            });
            wrapper.find(Message).vm.$emit('itemClicked');
            expect(stub).toBeCalled();
        })
    })
})