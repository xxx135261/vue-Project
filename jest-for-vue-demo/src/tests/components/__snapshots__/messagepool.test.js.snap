// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessagePool has expected html 1`] = `
<div
  class="pool"
>
  <ul />
</div>
`;
